opal/corelib/string/encoding.rb

```diff
@@ -232,7 +232,11 @@ class String
   `Opal.defineProperty(String.prototype, 'internal_encoding', #{Encoding::UTF_8})`
 
   def bytes
-    @bytes ||= each_byte.to_a
+    return @bytes.dup if @bytes
```

Comment:
`@bytes ||= each_byte.to_a` does not work when running is strict mode because `self.bytes` is not allowed. The following exception is thrown:

```
Exception: Cannot create property 'bytes' on string 'foo'
```

Reply:
I'm confused... `bytes` should be available on any String since we are using:

```
Opal.defineProperty(String.prototype, 'bytes', nil)
```

Reply:
OK I get it, the problem is that in strict mode we cannot assign a property on a `string`. 

```js
(function strict() { 'use strict'; 'foo'.bytes = '' })()
// Uncaught TypeError: Cannot create property 'bytes' on string 'foo'
```

```js
(function strict() { 'use strict'; new String('foo').bytes = '' })()
// OK
```